WireAuthentication/Sources/WireAuthenticationUI/Views/Registration/VerificationEmailCode/VerificationEmailCodeViewModel.swift

```diff
@@ -0,0 +1,213 @@
+        self.password = password
+        self.name = name
+        self.onFlowCompletion = onFlowCompletion
+        self.code = Array(repeating: "", count: numberOfDigits)
+        self.numberOfDigits = numberOfDigits
+    }
+
+    // MARK: - Actions
+
+    func handleInputReturningFocus(
+        _ newValue: String,
+        at index: Int
+    ) -> Int? {
+        if let intValue = Int(newValue.prefix(1)), (0 ... 9).contains(intValue) {
+            code[index] = String(intValue)
+        } else {
+            code[index] = ""
+        }
+
+        return if !code[index].isEmpty {
+            if index < numberOfDigits - 1 {
+                index + 1
+            } else {
+                nil
+            }
+        } else if index > 0 {
+            index - 1
+        } else {
+            0
+        }
+    }
+
+    private func register(verificationCode: String) async throws -> ([HTTPCookie], UUID?) {
+        let useCase = try await factory.registerPersonalAccountUseCase()
+        return try await Task.detached { [email, password, name] in
+            try await useCase.invoke(
+                email: email,
+                password: password,
+                verificationCode: verificationCode,
+                name: name
+            )
+        }.value
+    }
+
+    func confirm() async {
+        isLoading = true
+        let verificationCode = code.joined()
+        do {
+            let (cookies, uuid) = try await register(verificationCode: verificationCode)
+            guard let uuid else {
+                // add logs and show the error
+                return
+            }
+            let emailCredentials = EmailCredentials(
+                email: email,
+                password: password,
+                verificationCode: nil
+            )
+            let authenticationResult = try await createAuthenticationResult(
+                cookies: cookies,
+                emailCredentials: emailCredentials,
+                userID: uuid
+            )
+            onFlowCompletion(authenticationResult)
+        } catch {
+            WireLogger.authentication.error("register personal account failed: \(error)")
+
+            switch error {
+            case RegisterPersonalAccountUseCaseError.invalidEmail:
+                alert = .invalidEmailForRegistration
+            case RegisterPersonalAccountUseCaseError.blacklistedEmail:
+                alert = .blacklistedEmail
+            case RegisterPersonalAccountUseCaseError.tooManyTeamMembers:
+                alert = .tooManyTeamMembers
+            case RegisterPersonalAccountUseCaseError.userCreationRestricted:
+                alert = .userCreationRestricted
+            case RegisterPersonalAccountUseCaseError.invalidCode:
+                alert = .invalidCode
+            case RegisterPersonalAccountUseCaseError.emailExists:
+                alert = .emailExists
+            default:
+                router.presentAlert(for: error)
+            }
```

Comment:
suggestion: Let's not refactor it now, but in my opinion it would make more sense to use `LocalizedError` instead of mapping all possible error types to the correct strings.

Reply:
Noted!